File: backend/src/MomVibe.Domain/Entities/ChildFriendlyPlace.cs
namespace MomVibe.Domain.Entities;

using Common;
using Enums;

/// <summary>
/// Represents a child-friendly place submitted by a user, such as a playground, restaurant, or park.
/// Entries must be approved by an administrator before they become publicly visible.
/// </summary>
public class ChildFriendlyPlace : BaseEntity
{
    /// <summary>Gets or sets the identifier of the user who submitted this place.</summary>
    public string UserId { get; set; } = string.Empty;

    /// <summary>Gets or sets the user who submitted this place.</summary>
    public ApplicationUser User { get; set; } = null!;

    /// <summary>Gets or sets the display name of the place.</summary>
    public string Name { get; set; } = string.Empty;

    /// <summary>Gets or sets a detailed description of the place and its child-friendly features.</summary>
    public string Description { get; set; } = string.Empty;

    /// <summary>Gets or sets the street address of the place.</summary>
    public string? Address { get; set; }

    /// <summary>Gets or sets the city where the place is located.</summary>
    public string City { get; set; } = string.Empty;

    /// <summary>Gets or sets the category type of the place.</summary>
    public PlaceType PlaceType { get; set; }

    /// <summary>Gets or sets the minimum recommended child age in months.</summary>
    public int? AgeFromMonths { get; set; }

    /// <summary>Gets or sets the maximum recommended child age in months.</summary>
    public int? AgeToMonths { get; set; }

    /// <summary>Gets or sets the URL of a representative photo for the place.</summary>
    public string? PhotoUrl { get; set; }

    /// <summary>Gets or sets the official website URL of the place.</summary>
    public string? Website { get; set; }

    /// <summary>Gets or sets a value indicating whether an administrator has approved this submission.</summary>
    public bool IsApproved { get; set; }
}

File: frontend/src/types/childFriendlyPlace.ts
export const PlaceType = {
  Walk: 0,
  Playground: 1,
  Restaurant: 2,
  Cafe: 3,
  Museum: 4,
  Zoo: 5,
  Beach: 6,
  Park: 7,
  ThemeAttraction: 8,
  SportsActivity: 9,
  Other: 10,
} as const;

export type PlaceType = (typeof PlaceType)[keyof typeof PlaceType];

export interface ChildFriendlyPlaceDto {
  id: string;
  userId: string;
  authorDisplayName: string | null;
  name: string;
  description: string;
  address: string | null;
  city: string;
  placeType: PlaceType;
  ageFromMonths: number | null;
  ageToMonths: number | null;
  photoUrl: string | null;
  website: string | null;
  isApproved: boolean;
  createdAt: string;
}

export interface CreateChildFriendlyPlaceDto {
  name: string;
  description: string;
  address?: string;
  city: string;
  placeType: PlaceType;
  ageFromMonths?: number;
  ageToMonths?: number;
  photoUrl?: string;
  website?: string;
}
